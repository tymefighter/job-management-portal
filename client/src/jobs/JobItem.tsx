import "../styles/JobItem.scss";

interface JobItemProps {
    companyName: string;
    jobName: string;
    salary: number;
};

function JobItem({companyName, jobName, salary}: JobItemProps) {
    return (
        <div className="job-item">
            <span className="job-item__span">{jobName}</span>
            <span className="job-item__span">{companyName}</span>
            <span className="job-item__span">â‚¹{salary}</span>
        </div>
    );
}

export default JobItem;