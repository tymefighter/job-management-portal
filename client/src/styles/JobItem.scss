.job-item {
    display: flex;
    padding: 0.5rem 2rem;
    font-size: 1.5rem;
    background-color: white;
    border: 0.2rem solid black;
    justify-content: space-between;
}

.job-item__span {
    flex-grow: 0 30%;
    padding: 0 0.4rem;
}

@media screen and (max-width: 60rem) {
    .job-item {
        font-size: 1rem;
    }
}