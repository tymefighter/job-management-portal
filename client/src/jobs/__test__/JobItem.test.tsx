import { render } from "@testing-library/react";
import JobItem from "../JobItem";

test("JobItem content test", () => {
    const companyName = "This is a company name";
    const jobName = "This is a job name";
    const salary = 19384109823748;

    const {container} = render(
        <JobItem
            companyName={companyName} jobName={jobName}
            salary={salary}
        />
    );

    expect(container.textContent).toContain(companyName);
    expect(container.textContent).toContain(jobName);
    expect(container.textContent).toContain(salary);
});

test("JobItem snapshot test", () => {
    const companyName = "This is a company name";
    const jobName = "This is a job name";
    const salary = 19384109823748;

    const {container} = render(
        <JobItem
            companyName={companyName} jobName={jobName}
            salary={salary}
        />
    );

    expect(container).toMatchSnapshot("JobItem Component Snapshot");
});