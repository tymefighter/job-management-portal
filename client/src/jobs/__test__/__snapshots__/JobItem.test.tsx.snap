// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem snapshot test: JobItem Component Snapshot 1`] = `
<div>
  <div
    class="job-item"
  >
    <span
      class="job-item__span"
    >
      This is a job name
    </span>
    <span
      class="job-item__span"
    >
      This is a company name
    </span>
    <span
      class="job-item__span"
    >
      â‚¹
      19384109823748
    </span>
  </div>
</div>
`;
